### PR TITLE
Using get raw abi in cleos

### PR DESCRIPTION
**Pull request by allenhan2**
## Change Description

Replace get_abi with get_raw_abi to fetch abi file, then convert the raw abi to abi before sending back to the requestor.

## Consensus Changes
- [ ] Consensus Changes

## API Changes
- [ ] API Changes

## Documentation Additions
- [ ] Documentation Additions

### CODE REVIEW

**Comment by heifner** [comment]
Note: `get_raw_abi` was added in `v1.3.0` so this will break interacting with nodeos older than that version. Since all recent supported versions should work this is just an FYI.

File: programs/cleos/main.cpp [hunk 1]
```diff
@@ -2673,15 +2673,22 @@ int main( int argc, char** argv ) {
...
-         std::cerr << localized("saving abi to ${filename}", ("filename", filename)) << std::endl;
-         std::ofstream abiout( filename.c_str() );
-         abiout << abi;
-      } else {
-         std::cout << abi << "\n";
+      auto raw_abi_result = call(get_raw_abi_func, fc::mutable_variant_object("account_name", accountName));
+      auto raw_abi_blob = raw_abi_result["abi"].as_blob().data;
+      if (raw_abi_blob.size() != 0) {
+          auto abi_result = fc::raw::unpack<abi_def>(raw_abi_result["abi"].as_blob().data);
+          std::cout << "789\n";
```
Comment (jeffreyssmith2nd) [inline programs/cleos/main.cpp]: This looks like debugging code that should be removed.

File: programs/cleos/main.cpp [hunk 2]
```diff
@@ -445,13 +445,13 @@ auto abi_serializer_resolver = [](const name& account) -> fc::optional<abi_seria
...
-      if( abi_results.abi.valid() ) {
-         abis.emplace( *abi_results.abi, abi_serializer_max_time );
-      } else {
+       fc::optional<abi_serializer> abis;
+       auto raw_abi_result = call(get_raw_abi_func, fc::mutable_variant_object("account_name", account));
+       auto raw_abi_blob = raw_abi_result["abi"].as_blob().data;
+       if (raw_abi_blob.size() != 0) {
+           auto abi_result = fc::raw::unpack<abi_def>(raw_abi_blob);
+           abis.emplace(abi_result, abi_serializer_max_time);
+       } else {
```
Comment (jeffreyssmith2nd) [inline programs/cleos/main.cpp]: It's hard to tell on Github, but it looks like you changed the indentation here. In general, most of our code uses 3 spaces.

File: programs/cleos/main.cpp [hunk 3]
```diff
@@ -2673,15 +2673,22 @@ int main( int argc, char** argv ) {
...
+              std::cerr << localized("saving abi to ${filename}", ("filename", filename)) << std::endl;
+              std::ofstream abiout(filename.c_str());
+              abiout << abi;
+          } else {
+              std::cout << abi << "\n";
+          }
+      } else
+      {
+        FC_THROW_EXCEPTION(key_not_found_exception, "Key ${key}", ("key", "abi"));
       }
```
Comment (heifner) [inline programs/cleos/main.cpp]: Please follow style you see used in existing code. `} else {`

File: programs/cleos/main.cpp (hunk 2 above)
Comment (heifner) [reply programs/cleos/main.cpp]: Make sure you are not using tabs.

**Pull request merged**